lib/inspec/plugin/v2/installer.rb

```diff
@@ -284,8 +290,16 @@ def install_from_gem_file(requested_plugin_name, opts)
 
     def install_from_remote_gems(requested_plugin_name, opts)
       plugin_dependency = Gem::Dependency.new(requested_plugin_name, opts[:version] || "> 0")
-      # BestSet is rubygems.org API + indexing
-      install_gem_to_plugins_dir(plugin_dependency, [Gem::Resolver::BestSet.new], opts[:update_mode])
+
+      # BestSet is rubygems.org API + indexing, APISet is for custom sources
+      sources = Gem::Resolver::BestSet.new
+      sources = Gem::Resolver::APISet.new(opts[:source]) if opts[:source]
+
+      install_gem_to_plugins_dir(plugin_dependency, [sources], opts[:update_mode])
+    rescue TypeError
+      # Retry with appending the dependencies endpoint
+      sources = Gem::Resolver::APISet.new(opts[:source] + "/api/v1/dependencies")
```

Comment:
Is this always a required part of the URL? If so, could it be checked for and appended if missing, rather than using trying and failing?

Reply:
From what I could see in the Rubygems API, they are placing that as a default which makes me assume it's a required part. Do you propose switching the two around or should I rather implement some sort of preflight check using Net::HTTP instead?

Reply:
I meant doing a string check (likely by parsing it with URI) on the provided URL, and if it does not contain the right path, then append the correct path.